src/python/pants/backend/python/providers/pbs/rules.py

```diff
@@ -0,0 +1,237 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+import collections.abc
+import functools
+import json
+import os
+import textwrap
+from typing import Iterable, TypedDict, cast
+
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
+from pants.backend.python.util_rules.pex import PythonProvider
+from pants.backend.python.util_rules.pex import rules as pex_rules
+from pants.backend.python.util_rules.pex_environment import PythonExecutable
+from pants.core.util_rules.external_tool import (
+    DownloadedExternalTool,
+    ExternalToolError,
+    ExternalToolRequest,
+)
+from pants.core.util_rules.external_tool import rules as external_tools_rules
+from pants.core.util_rules.system_binaries import CpBinary
+from pants.engine.fs import DownloadFile
+from pants.engine.internals.native_engine import FileDigest
+from pants.engine.internals.selectors import Get
+from pants.engine.platform import Platform
+from pants.engine.process import Process, ProcessCacheScope, ProcessResult
+from pants.engine.rules import collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.option.global_options import NamedCachesDirOption
+from pants.option.option_types import StrListOption
+from pants.option.subsystem import Subsystem
+from pants.util.docutil import bin_name
+from pants.util.frozendict import FrozenDict
+from pants.util.logging import LogLevel
+from pants.util.resources import read_sibling_resource
+from pants.util.strutil import softwrap
+
+PBS_SANDBOX_NAME = ".pbs"
+PBS_NAMED_CACHE_NAME = "pbs"
+PBS_APPEND_ONLY_CACHES = FrozenDict({PBS_NAMED_CACHE_NAME: PBS_SANDBOX_NAME})
+
+
+class PBSPythonInfo(TypedDict):
+    url: str
+    sha256: str
+    size: int
+
+
+@functools.cache
+def load_pbs_pythons() -> dict[str, dict[str, PBSPythonInfo]]:
+    return cast(
+        "dict[str, dict[str, PBSPythonInfo]]",
+        json.loads(read_sibling_resource(__name__, "versions_info.json"))["pythons"],
+    )
+
+
+class PBSPythonProviderSubsystem(Subsystem):
+    options_scope = "pbs-python-provider"
+    name = "pbs"
```

Comment:
As far as I can tell, "PBS" seems to be a Pex/Pants-ism, and the project itself refers to itself as "Python Standalone Builds" (i.e. would shorten to PSB), with only repo name/URL slugs being "python-build-standalone". Searching for "python PBS" doesn't seem to find it, either.

Given the options scope, backend name etc. are rarely written (and other things like the named cached dir are internal, but user visible to someone investigating issues), I'd personally lean towards spelling them out, e.g.

```suggestion
    options_scope = "python-build-standalone-python-provider"
    name = "python-build-standalone"
```

And similarly for the named cache dir/sandbox names.

(Using "PBS" in the doc string is fine, as long as it's explained the first time, as you already do 👍 )